lnd/client.go

```diff
@@ -230,3 +269,41 @@ func (c *LndClient) getWaitingCloseChannels(nodeID string) ([]*lnrpc.PendingChan
 	}
 	return waitingCloseChannels, nil
 }
+
+func (c *LndClient) WaitOnline(peerID []byte, timeout time.Time) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	pkStr := hex.EncodeToString(peerID)
+	sub, err := c.client.SubscribePeerEvents(ctx, &lnrpc.PeerEventSubscription{})
+	if err != nil {
+		return err
+	}
+
+	// NOTE: It would be nice to have a lookup for a single peer in LND.
+	peers, err := c.client.ListPeers(ctx, &lnrpc.ListPeersRequest{})
+	if err != nil {
+		return err
+	}
+
+	for _, peer := range peers.Peers {
+		if peer.PubKey == pkStr {
+			return nil
+		}
+	}
+
+	for {
```

Comment:
There is a small gap between this loop and the listpeers above so in theory we may miss the notification.

Reply:
The peer events subscription is before the ListPeers call, so I was thinking if an update happens before/during the ListPeers call, we'll catch it in that first Recv.

Reply:
Right!